File: frontend/src/constants/trade.ts
export const TRADE_STATUSES = [
  { value: 'pending', label: '待确认', type: 'warning' },
  { value: 'confirmed', label: '已确认', type: 'primary' },
  { value: 'completed', label: '已完成', type: 'success' },
  { value: 'cancelled', label: '已取消', type: 'info' },
] as const

export const REVIEW_RATINGS = [
  { value: 'good', label: '好评' },
  { value: 'medium', label: '中评' },
  { value: 'bad', label: '差评' },
] as const

export const APPOINTMENT_STATUSES = [
  { value: 'pending', label: '待响应', type: 'warning' },
  { value: 'accepted', label: '已接受', type: 'primary' },
  { value: 'rejected', label: '已拒绝', type: 'danger' },
  { value: 'superseded', label: '已改约', type: 'info' },
  { value: 'completed', label: '已完成', type: 'success' },
  { value: 'cancelled', label: '已取消', type: 'info' },
] as const

export function appointmentStatusLabel(value: string): string {
  return APPOINTMENT_STATUSES.find((s) => s.value === value)?.label ?? value
}

export function appointmentStatusType(value: string): string {
  return APPOINTMENT_STATUSES.find((s) => s.value === value)?.type ?? 'info'
}

export function tradeStatusLabel(value: string): string {
  return TRADE_STATUSES.find((t) => t.value === value)?.label ?? value
}

export function tradeStatusType(value: string): string {
  return TRADE_STATUSES.find((t) => t.value === value)?.type ?? 'info'
}

export function ratingLabel(value: string): string {
  return REVIEW_RATINGS.find((r) => r.value === value)?.label ?? value
}

File: backend/internal/constants/trade.go
package constants

// TradeStatus defines trade order state machine values shared with the frontend.
const (
	TradeStatusPending   = "pending"
	TradeStatusConfirmed = "confirmed"
	TradeStatusCompleted = "completed"
	TradeStatusCancelled = "cancelled"
)

// TradeStatuses lists all valid trade statuses in flow order.
var TradeStatuses = []string{
	TradeStatusPending, TradeStatusConfirmed, TradeStatusCompleted, TradeStatusCancelled,
}

// IsTradeStatus reports whether the given status is valid.
func IsTradeStatus(s string) bool {
	for _, v := range TradeStatuses {
		if v == s {
			return true
		}
	}
	return false
}

// TradeStatusText returns the Chinese label of a trade status.
func TradeStatusText(s string) string {
	switch s {
	case TradeStatusPending:
		return "待确认"
	case TradeStatusConfirmed:
		return "已确认"
	case TradeStatusCompleted:
		return "已完成"
	case TradeStatusCancelled:
		return "已取消"
	default:
		return "未知"
	}
}

// AppointmentStatus defines meetup appointment state machine values shared with the frontend.
const (
	AppointmentStatusPending    = "pending"
	AppointmentStatusAccepted   = "accepted"
	AppointmentStatusRejected   = "rejected"
	AppointmentStatusSuperseded = "superseded"
	AppointmentStatusCompleted  = "completed"
	AppointmentStatusCancelled  = "cancelled"
)

// AppointmentStatuses lists all valid appointment statuses in flow order.
var AppointmentStatuses = []string{
	AppointmentStatusPending, AppointmentStatusAccepted, AppointmentStatusRejected,
	AppointmentStatusSuperseded, AppointmentStatusCompleted, AppointmentStatusCancelled,
}

// AppointmentActiveStatuses lists the statuses that count as the single valid
// appointment an order may hold at any moment.
var AppointmentActiveStatuses = []string{
	AppointmentStatusPending, AppointmentStatusAccepted,
}

// IsAppointmentStatus reports whether the given status is valid.
func IsAppointmentStatus(s string) bool {
	for _, v := range AppointmentStatuses {
		if v == s {
			return true
		}
	}
	return false
}

// IsAppointmentActive reports whether the status blocks a new appointment.
func IsAppointmentActive(s string) bool {
	for _, v := range AppointmentActiveStatuses {
		if v == s {
			return true
		}
	}
	return false
}

// AppointmentStatusText returns the Chinese label of an appointment status.
func AppointmentStatusText(s string) string {
	switch s {
	case AppointmentStatusPending:
		return "待响应"
	case AppointmentStatusAccepted:
		return "已接受"
	case AppointmentStatusRejected:
		return "已拒绝"
	case AppointmentStatusSuperseded:
		return "已改约"
	case AppointmentStatusCompleted:
		return "已完成"
	case AppointmentStatusCancelled:
		return "已取消"
	default:
		return "未知"
	}
}

// ReviewRating defines review rating enum values shared with the frontend.
const (
	ReviewRatingGood   = "good"
	ReviewRatingMedium = "medium"
	ReviewRatingBad    = "bad"
)

// ReviewRatingText returns the Chinese label of a review rating.
func ReviewRatingText(r string) string {
	switch r {
	case ReviewRatingGood:
		return "好评"
	case ReviewRatingMedium:
		return "中评"
	case ReviewRatingBad:
		return "差评"
	default:
		return "未知"
	}
}
